### PR TITLE
feat: tsx support

### LINKED ISSUE #476: [🚀tsx]: request for correct out of the box TSX tags definitions
As an ESL consumer, I want to be able to use ESL custom tags with TSX without manual tags shape definition.

Proposed:
- [x] add out of the box ESL components TSX definitions
- [x] add complete Base element shape definition
- [x] should be compatible with `jsx-dom` library
- [x] fix known shape definitions problems
- [x] add definitions for the missed stable components
- [x] support ARIA atributes

**Priority**: Normal

**Additional context**
Be aware of the base shape definition problem, it is proposed to use ESL own one to be sure in compatibility.
Use compatible TSX `IntrinsicElements` interface expansion

### PR DESCRIPTION
**Pull request by ala-n**
Resolves #476

### CODE REVIEW

File: src/modules/esl-base-element/shape/wai-aria.shape.ts [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+  /**
+   * Indicates whether inputting text could trigger display of one or more predictions of the user's intended value
+   * for an input and specifies how predictions would be presented if they are made.
+   */
+  'aria-autocomplete'?: 'none' | 'inline' | 'list' | 'both';
+
+  /**
+   * Indicates an element is being modified and that assistive technologies MAY want to wait until
+   * the modifications are complete before exposing them to the user.
```
Comment (yadamskaya) [inline src/modules/esl-base-element/shape/wai-aria.shape.ts:27]: 'are completed'

File: src/modules/esl-base-element/shape/wai-aria.shape.ts [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+   */
+  'aria-dropeffect'?: 'none' | 'copy' | 'execute' | 'link' | 'move' | 'popup';
+
+  /**
+   * Identifies the element that provides an error message for the object.
+   * @see aria-invalid @see aria-describedby.
+   */
+  'aria-errormessage'?: string;
+
+  /** Indicates whether the element, or another grouping element it controls, is currently expanded or collapsed. */
```
Comment (yadamskaya) [inline src/modules/esl-base-element/shape/wai-aria.shape.ts:94]: You can remove unnecessary commas

File: src/modules/esl-alert/core/esl-alert.shape.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import type {ESLToggleableTagShape} from '../../esl-toggleable/core/esl-toggleable.shape';
+import type {ESLAlert} from './esl-alert';
+
+/**
+ * Tag declaration interface of {@link ESLAlert} element
+ * Used for TSX declaration
+ */
+export interface ESLAlertShape extends ESLToggleableTagShape<ESLAlert> {
+  /**
+   * Define the scope (using {@link TraversingQuery} syntax) element to listen to activation event.
```
Comment (yadamskaya) [inline src/modules/esl-alert/core/esl-alert.shape.ts:10]: 'Defines the scope'?

**Review by yadamskaya — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

File: src/modules/esl-base-element/shape/wai-aria.shape.ts [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+  'aria-expanded'?: 'true' | 'false' | boolean;
+
+  /**
+   * Identifies the next element (or elements) in an alternate reading order of content which, at the user's discretion,
+   * allows assistive technology to override the general default of reading in document source order.
```
[the remark this answers is not in the document]
Comment (ala-n) [reply src/modules/esl-base-element/shape/wai-aria.shape.ts:99]: Add an article but left  `override the general default of reading`

File: src/modules/esl-base-element/shape/wai-aria.shape.ts [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+   * Identifies the element (or elements) that labels the current element.
+   * @see aria-describedby.
+   */
+  'aria-labelledby'?: string;
+
+  /** Defines the hierarchical level of an element within a structure. */
+  'aria-level'?: number;
+
+  /**
+   * Indicates that an element will be updated, and describes the types of updates the user agents,
```
[the remark this answers is not in the document]
Comment (ala-n) [reply src/modules/esl-base-element/shape/wai-aria.shape.ts:144]: Rephrased

**Review by andreybelous — APPROVED** [review]

**Review by julia-murashko — APPROVED** [review]

**Pull request merged**